util/src/migration/tests.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::BTreeMap;
+use migration::{Error, Destination, Migration, Manager, Config};
+
+impl Destination for BTreeMap<Vec<u8>, Vec<u8>> {
+	fn commit(&mut self, batch: BTreeMap<Vec<u8>, Vec<u8>>) -> Result<(), Error> {
+		self.extend(batch);
+		Ok(())
+	}
+}
+
+struct Migration0;
+
+impl Migration for Migration0 {
+	fn from_version(&self) -> &'static str {
+		"0"
```

Comment:
Might be worth to change version from string to enum.
Benefits:
- All possible versions in single place (might include additional comments / meta information)
- Pattern matching all possible versions is possible.
- Can be ordered.
- You can enforce that migrations exists for all versions.